src/main/java/io/jenkins/plugins/sprp/GitPushStep.java

```diff
@@ -0,0 +1,106 @@
+package io.jenkins.plugins.sprp;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.Queue;
+import hudson.model.TaskListener;
+import hudson.model.queue.Tasks;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.Collections;
+import java.util.Set;
+
+public class GitPushStep extends Step {
```

Comment:
So it becomes a generic step. Fine with me, but maybe we want to move it to another plugin (e.g. Git Client) later. CC @MarkEWaite

Reply:
Will create a Jira for this.